mcs/class/corlib/System/AppDomainSetup.cs

```diff
@@ -139,14 +140,19 @@ static string GetAppBase (string appBase)
 				appBase = appBase.Substring (7);
 				if (Path.DirectorySeparatorChar != '/')
 					appBase = appBase.Replace ('/', Path.DirectorySeparatorChar);
-				if (Environment.IsRunningOnWindows) {
-					// Under Windows prepend "//" to indicate it's a local file
-					appBase = "//" + appBase;
+			}
+			appBase = Path.GetFullPath (appBase);
+
+			if (Path.DirectorySeparatorChar != '/') {
+				bool isExtendedPath = appBase.StartsWith (@"\\?\", StringComparison.Ordinal);
+				if (appBase.IndexOf (':', isExtendedPath ? 6 : 2) != -1) {
+					throw new NotSupportedException ("The given path's format is not supported.");
 				}
-			} else {
-				appBase = Path.GetFullPath (appBase);
 			}
 
+			// Trigger a validation of the path
+			new FileIOPermission (FileIOPermissionAccess.PathDiscovery, appBase);
```

Comment:
I see. You might have missed it but I removed the call to `Demand()` so AFAICT no security check is performed. The only thing I'm interested in is invoking the FileIOPermission() constructor which validates the path characters. Are you saying that the FileIOPermission class is unavailable unless FEATURE_MONO_CAS is defined? On Windows FEATURE_MONO_CAS doesn't seem to be defined but invoking the constructor seems to work just fine.